### PR TITLE
WPC-202 rich-text Get characters count method, and get string text method

### PR DESCRIPTION
**Pull request by bishoymelek**
**Characters Count:**
- Get characters' count method
- Show/Hide prop for the count.

**String Text:**
- Get string text method.
- Get string text method as a prop

**Note: it should be reviewed after merging f/WPC-212**

### CODE REVIEW

File: packages/editor/src/client/blocks/richTextBlock/richTextBlock.tsx [hunk 1]
```diff
@@ -15,14 +15,24 @@ import {withNormalizeNode} from './editor/normalizing'
 import {LinkFormatButton, RemoveLinkFormatButton} from './toolbar/linkButton'
 import {TableMenu} from './toolbar/tableMenu'
 import {WepublishEditor} from './editor/wepublishEditor'
+import {Alert} from 'rsuite'
 
-export type RichTextBlockProps = BlockProps<RichTextBlockValue>
+export type RichTextBlockProps = BlockProps<RichTextBlockValue> & {
+  maxCharsLimit?: number
+  getCharsLength?: any
```
Comment (faebebin) [inline packages/editor/src/client/blocks/richTextBlock/richTextBlock.tsx:22]: I think the type shoud be: `(text: string) => number` .
But I do not really see why this method is necessary at all?

Comment (faebebin) [inline packages/editor/src/client/blocks/richTextBlock/richTextBlock.tsx:22]: Could this have a type?

**Review by faebebin — COMMENTED** [review]
It looks good, but as it is not yet in use anywhere? it is a bit hard to say how it really behaves ...

File: packages/editor/src/client/blocks/richTextBlock/richTextBlock.tsx [hunk 2]
```diff
@@ -18,14 +18,18 @@ import {WepublishEditor} from './editor/wepublishEditor'
...
 }
 
 export const RichTextBlock = memo(function RichTextBlock({
   value,
   autofocus,
   disabled,
   onChange,
-  displayOnly = false
+  displayOnly = false,
+  showCharCount = true,
```
Comment (roosnic1) [inline packages/editor/src/client/blocks/richTextBlock/richTextBlock.tsx:31]: default should be `false`.

File: packages/editor/src/client/blocks/richTextBlock/richTextBlock.tsx [hunk 3]
```diff
@@ -18,14 +18,18 @@ import {WepublishEditor} from './editor/wepublishEditor'
 
 export interface RichTextBlockProps extends BlockProps<RichTextBlockValue> {
   displayOnly?: boolean
+  showCharCount?: boolean
+  getStringText?: (text: string) => void
```
Comment (roosnic1) [inline packages/editor/src/client/blocks/richTextBlock/richTextBlock.tsx:22]: Not needed.

File: packages/editor/src/client/blocks/richTextBlock/richTextBlock.tsx [hunk 4]
```diff
@@ -122,6 +133,9 @@ export const RichTextBlock = memo(function RichTextBlock({
           setLocation(editor.selection)
         }}
       />
+      {showCharCount && (
+        <p style={{textAlign: 'right'}}>{`${t('blocks.richText.charsCount')}: ${charCount}`}</p>
```
Comment (roosnic1) [inline packages/editor/src/client/blocks/richTextBlock/richTextBlock.tsx:137]: https://www.i18next.com/translation-function/interpolation

File: packages/editor/src/client/blocks/richTextBlock/editor/wepublishEditor.ts [hunk 5]
```diff
@@ -58,5 +58,17 @@ export const WepublishEditor = {
...
   isEmpty(editor: Editor) {
     return JSON.stringify(editor.children) === JSON.stringify(this.createDefaultValue())
+  },
+
+  calculateEditorCharCount(editor: Editor) {
+    return this.getTextString(editor).length
+  },
+
+  getTextString(editor: Editor) {
+    return [...Node.texts(editor)].reduce((string, nodePair, index) => {
```
Comment (roosnic1) [inline packages/editor/src/client/blocks/richTextBlock/editor/wepublishEditor.ts:68]: maybe add a little comment what exactly you are doing here.

File: packages/api/__tests__/api/public/index.ts [hunk 6]
```diff
@@ -228,6 +228,29 @@ export type InstagramPostBlock = {
   postID: Scalars['String']
 }
 
+export type Invoice = {
```
Comment (roosnic1) [inline packages/api/__tests__/api/public/index.ts:231]: Why is this file in the PR?

File: packages/api/__tests__/api/private/index.ts [hunk 7]
```diff
@@ -1199,6 +1199,7 @@ export type QueryUsersArgs = {
   before?: Maybe<Scalars['ID']>
   first?: Maybe<Scalars['Int']>
   last?: Maybe<Scalars['Int']>
+  skip?: Maybe<Scalars['Int']>
```
Comment (roosnic1) [inline packages/api/__tests__/api/private/index.ts:1202]: why is this file in the PR?

**Review by roosnic1 — CHANGES_REQUESTED** [review]

File: packages/editor/src/client/blocks/richTextBlock/editor/wepublishEditor.ts (hunk 5 above)
Comment (faebebin) [reply packages/editor/src/client/blocks/richTextBlock/editor/wepublishEditor.ts:68]: This solution looks good. Would be nice to have some unittests on this.

**Review by faebebin — CHANGES_REQUESTED** [review]

File: packages/editor/src/client/blocks/richTextBlock/editor/wepublishEditor.ts (hunk 5 above)
Comment (bishoymelek) [reply packages/editor/src/client/blocks/richTextBlock/editor/wepublishEditor.ts:69]: @faebebin Do we already have unit tests for this block?

File: packages/editor/src/client/blocks/richTextBlock/richTextBlock.tsx (hunk 3 above)
Comment (bishoymelek) [reply packages/editor/src/client/blocks/richTextBlock/richTextBlock.tsx]: Why?

File: packages/editor/src/client/blocks/richTextBlock/editor/wepublishEditor.ts (hunk 5 above)
Comment (faebebin) [reply packages/editor/src/client/blocks/richTextBlock/editor/wepublishEditor.ts:69]: No. But would be good to have IMO, there might be more methods in the future.

Comment (faebebin) [reply packages/editor/src/client/blocks/richTextBlock/editor/wepublishEditor.ts:69]: @bishoymelek there are no unittests yet, but you could have a look a the normalizing.test.ts. 
I would create some editor instance with diefferent content and then invoke the method to assert the string.
Would be good to have this as template for future methods also.

Maybe something like:
expect(emptyParagraph).toMatch('')
expect(... text: ' '}).toMatch(' ')
expect(...{ text: 'hello' }).toMatch('hello')
expect(... complex block with tables and lists but not text ).toMatch('')
expect(... complex block with tables and lists but not text ).toMatch('concated text')

File: packages/editor/src/client/blocks/richTextBlock/richTextBlock.tsx (hunk 3 above)
Comment (roosnic1) [reply packages/editor/src/client/blocks/richTextBlock/richTextBlock.tsx]: because we don't need the functionality in the block.

**Review by HanaAyman — APPROVED** [review]

**Review by 0815pascal — APPROVED** [review]

File: packages/editor/src/client/blocks/richTextBlock/editor/wepublishEditor.ts (hunk 5 above)
Comment (bishoymelek) [reply packages/editor/src/client/blocks/richTextBlock/editor/wepublishEditor.ts:71]: @faebebin Will make separate PR

**Review by roosnic1 — APPROVED** [review]

**Review by faebebin — APPROVED** [review]

**Pull request merged**